truss-chains/truss_chains/stub.py

```diff
@@ -55,12 +59,19 @@ def predict_sync(self, json_payload):
                     logging.info(
                         f"Retrying `{self._service_descriptor.name}`, " f"attempt {num}"
                     )
-                return utils.handle_response(
-                    self._client_sync.post(
-                        self._service_descriptor.predict_url, json=json_payload
-                    ),
-                    self.name,
-                )
+                with tracer.start_as_current_span(name="predict_remote") as span:
+                    span.set_attribute("remote_model", self._service_descriptor.name)
+                    headers = {}
+                    ctx = context.get_current()
+                    TraceContextTextMapPropagator().inject(headers, ctx) 
+                    response = utils.handle_response(
+                        self._client_sync.post(
+                            self._service_descriptor.predict_url, json=json_payload, headers=headers
```

Comment:
Do the headers get merged/updated?

Reply:
yes, for each request the header is updated